src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -348,20 +351,8 @@ FindShardIntervalIndex(Datum searchedValue, CitusTableCacheEntry *cacheEntry)
 		else
 		{
 			int hashedValue = DatumGetInt32(searchedValue);
-			uint64 hashTokenIncrement = HASH_TOKEN_COUNT / shardCount;
-
-			shardIndex = (uint32) (hashedValue - INT32_MIN) / hashTokenIncrement;
-			Assert(shardIndex <= shardCount);
 
-			/*
-			 * If the shard count is not power of 2, the range of the last
-			 * shard becomes larger than others. For that extra piece of range,
-			 * we still need to use the last shard.
-			 */
-			if (shardIndex == shardCount)
-			{
-				shardIndex = shardCount - 1;
-			}
+			shardIndex = CalculateUniformHashRangeIndex(hashedValue, shardCount);
```

Comment:
maybe/optional: explicitly cast to `int64` here otherwise we're passing `int` as `int64` and move the relevant comment here:
```
/* we use 64-bit integers to avoid overflow issues during arithmetic */
shardIndex = CalculateUniformHashRangeIndex((int64) hashedValue, shardCount);
```

Reply:
I changed the function signature to only accept `int`, and then do an explicit cast in the function.